hive-hooks/path-conversion-hook/src/main/java/com/expediagroup/apiary/extensions/hooks/pathconversion/converters/GenericConverter.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2018-2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.apiary.extensions.hooks.pathconversion.converters;
+
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+
+import lombok.extern.slf4j.Slf4j;
+
+import com.expediagroup.apiary.extensions.hooks.pathconversion.config.Configuration;
+
+@Slf4j
+public abstract class GenericConverter {
+
+    private final Configuration configuration;
+
+    public GenericConverter(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
+    public boolean convertPath(Table table) {
```

Comment:
why this has to return true/false? Maybe add an interface or some Javadoc on the public methods here explaining the contract? If I extend this and implement a subclass when should I return true and when return false?